registry-image-sync/action.yaml

```diff
@@ -0,0 +1,38 @@
+name: "Sync images between two registries"
+description: "sync images between two registries using skopeo"
+
+inputs:
+  SOURCE_REGISTRY:
+    description: "the source registry"
+    required: true
+  TARGET_REGISTRY:
+    description: "the target registry"
+    required: true
+  IMAGE_NAME:
+    description: "the name of the image to sync"
+  SOURCE_REPO:
+    description: "the source repo that contains the images"
+  SRC_USERNAME:
+    description: "the username in the source registry"
+  SRC_PASSWORD:
+    description: "the password in source registry"
+  DEST_USERNAME:
+    description: "the username in destination registry"
+  DEST_PASSWORD:
+    description: "the password in destination registry"
+
+
+runs:
+  using: "composite"
```

Comment:
Instead of making it a composite action, make it [a docker one](https://docs.github.com/en/actions/creating-actions/metadata-syntax-for-github-actions#runs-for-docker-container-actions). Removing the need to ask users to use the proper container image and giving you more control over the environment.